### ISSUE TITLE
Fix S2219 FP: "Use the is operator" reports "unfixable" code

### ISSUE THREAD

**Issue by jaens** [SonarSource/sonar-dotnet#6616]
### Description

When using a generic type without a generic type parameter (`Generic<>`), the S2219 rule triggers, but the suggested fix is impossible, as `object is Generic<>` is not valid C#.

### Repro steps

```csharp
var _ = typeof(ISet<>).IsInstanceOfType(obj);
```

(note: this code is probably nonsensical as it will always return `false`, since open generic types do not have instances)

### Expected behavior

Rule does not trigger. (or triggers with some other explanation)

### Actual behavior

Rule triggers.

Autofix creates invalid code:
```csharp
var _ = obj is ISet<>;
```

### Known workarounds

Disable the rule.

**Comment by pavel-mikula-sonarsource** [comment]
Hi @jaens,

Thank you for reporting this, I confirm it as a False Positive
